test/libsolidity/smtCheckerTests/out_of_bounds/array_2d_4.sol

```diff
@@ -24,4 +24,4 @@ contract C {
 // Info 1391: CHC: 1 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
 // Warning 2661: (184-197): BMC: Overflow (resulting value larger than 2**256 - 1) happens here.
 // Warning 2661: (228-244): BMC: Overflow (resulting value larger than 2**256 - 1) happens here.
-// Info 6002: BMC: 2 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
+// Info 6002: BMC: 4 verification condition(s) proved safe! Enable the model checker option "show proved safe" to see all of them.
```

Comment:
I wanted to see what has changed and ran:
`solc --model-checker-engine bmc --model-checker-solvers z3 --model-checker-show-proved-safe` on this contract but I am not getting any output except `Warning: This is a pre-release version...` @leonardoalt cd ..

Reply:
maybe it's underflow/overflow targets? these are disabled by default in `solc`, but enabled by the smtchecker tests in isoltest

Reply:
Right @leonardoalt 
Results compared:
```
2d_4.sol 
Warning: This is a pre-release compiler version, please do not use it in production.

Warning: BMC: Overflow (resulting value larger than 2**256 - 1) happens here.
  --> array_2d_4.sol:11:24:
   |
11 | 		for (uint i = 0; i < a.length + 10; ++i)
   | 		                     ^^^^^^^^^^^^^
Note: Counterexample:
  <result> = 2**256
  a.length = 0xFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFfff6
  i = 0
  j = 0

Note: Callstack:
Note: False negatives are possible when unrolling loops.
This is due to the possibility that the BMC loop iteration setting is smaller than the actual number of iterations needed to complete a loop.

Warning: BMC: Overflow (resulting value larger than 2**256 - 1) happens here.
  --> array_2d_4.sol:12:25:
   |
12 | 			for (uint j = 0; j < a[i].length + 20; ++j)
   | 			                     ^^^^^^^^^^^^^^^^
Note: Counterexample:
  <result> = 2**256
  a.length = 0
  a[i].length = 0xFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFffffFFFFffec
  i = 0
  j = 0

Note: Callstack:
Note: False negatives are possible when unrolling loops.
This is due to the possibility that the BMC loop iteration setting is smaller than the actual number of iterations needed to complete a loop.

Info: BMC: Overflow (resulting value larger than 2**256 - 1) check is safe!
  --> array_2d_4.sol:11:24:
   |
11 | 		for (uint i = 0; i < a.length + 10; ++i)
   | 		                     ^^^^^^^^^^^^^

Info: BMC: Overflow (resulting value larger than 2**256 - 1) check is safe!
  --> array_2d_4.sol:11:39:
   |
11 | 		for (uint i = 0; i < a.length + 10; ++i)
   | 		                                    ^^^

Info: BMC: Overflow (resulting value larger than 2**256 - 1) check is safe!
  --> array_2d_4.sol:12:25:
   |
12 | 			for (uint j = 0; j < a[i].length + 20; ++j)
   | 			                     ^^^^^^^^^^^^^^^^

Info: BMC: Overflow (resulting value larger than 2**256 - 1) check is safe!
  --> array_2d_4.sol:12:43:
   |
12 | 			for (uint j = 0; j < a[i].length + 20; ++j)
   | 			                                       ^^^
```
seems right

Reply:
makes sense!